src/hooks/EtherPrice.ts

```diff
@@ -1,59 +1,32 @@
-import { Contract } from '@ethersproject/contracts'
-import { readNetwork } from 'constants/networks'
-import { readProvider } from 'constants/readProvider'
-import { NetworkName } from 'models/network-name'
-import { useEffect, useState } from 'react'
+import { CV_V3 } from 'constants/cv'
+import { WAD_DECIMALS } from 'constants/numbers'
+import { V2V3ContractName } from 'models/v2v3/contracts'
+import { useEffect } from 'react'
 import { fromWad } from 'utils/format/formatNumber'
+import { V2V3_CURRENCY_ETH, V2V3_CURRENCY_USD } from 'utils/v2v3/currency'
 import { useContractReadValue } from './ContractReader'
-import { loadV1Contract } from './v1/V1ContractLoader'
+import { useLoadV2V3Contract } from './v2v3/LoadV2V3Contract'
 
 /**
  * Chainlink feed doesn't tend to up date that quickly.
  * Refresh every 5 minutes.
  */
 const PRICE_REFRESH_INTERVAL = 60 * 1000 * 5 // 5 minutes
 
-const usePricesContract = () => {
-  const [contract, setContract] = useState<Contract | undefined>()
-
-  useEffect(() => {
-    if (contract) return
-
-    const load = async () => {
-      try {
-        const contract = await loadV1Contract(
-          'Prices',
-          readNetwork.name,
-          readProvider,
-        )
-
-        setContract(contract)
-      } catch (_) {
-        // if Prices contract isn't deployed on the readNetwork,
-        // fall back to mainnet
-        const contract = await loadV1Contract(
-          'Prices',
-          NetworkName.mainnet,
-          readProvider,
-        )
-
-        setContract(contract)
-      }
-    }
-
-    load()
-  }, [contract])
-
-  return contract
-}
-
+/**
+ * Return the current price of ETH in USD.
+ * @example 1234.69
+ */
 export function useEtherPrice() {
-  const contract = usePricesContract()
+  const JBPrices = useLoadV2V3Contract({
```

Comment:
Load this contract specifically here, instead of relying on context. Means it'll work on V1 projects, where the V2V3 contracts context isn't available.